VisualPinball.Unity/VisualPinball.Unity/VPT/ItemBehavior.cs

```diff
@@ -102,7 +112,16 @@ public void RebuildMeshes()
 		public virtual Vector3 GetEditorScale() { return Vector3.zero; }
 		public virtual void SetEditorScale(Vector3 rot) { }
 
-		public bool IsLocked { get { return data.IsLocked; } set { data.IsLocked = value; } }
+		public virtual void HandleMaterialRenamed(string undoName, string oldName, string newName) { }
+
+		// rename helper to cut down on the boiler plate in the concrete classes
+		protected void TryRenameField(string undoName, ref string field, string oldName, string newName)
+		{
+			if (field == oldName) {
+				Undo.RecordObject(this, undoName);

```

Comment:
My IDE doesn't know `Undo`. Do you know from which DLL it comes from?

Reply:
This was another case of editor assembly stuff, I wrapped it in an ifdef to fix. I know its not ideal to have editor code in here (though to be fair all that stuff around there is, even if it doesn't live in that assembly). I'm sure we could refactor this out, but it would add a lot of complexity plumbing results all the back out to the callers etc.

Reply:
Okay for me!